### PR TITLE
docs: zephyr: Add note on swap-using-scratch

### PR DESCRIPTION
**Pull request by nordicjm**
Adds details on how to use the swap-using-scratch algorithm and associated risks.

### CODE REVIEW

**Comment by nordicjm** [comment]
@d3zd3z can this be reviewed please?

File: docs/readme-zephyr.md
Comment (utzig) [inline docs/readme-zephyr.md:211]: Is this related to #1165? Was anyone else able to reproduce it or you were the only person that has ever seen the error? Does it also happen when using only internal flash?

Comment (nordicjm) [reply docs/readme-zephyr.md:211]: Everyone at the company I worked at on multiple boards (pinnacle 100, bt610), people on discord with varying boards. Both boards I've seen it on had primary slots in internal flash, with secondary and scratch partitions in external QSPI memory (all devices with 4KB sector sizes)

Comment (utzig) [reply docs/readme-zephyr.md:211]: Pinnacle 100 and bt610 are also nrf52840, correct? Do they also use QSPI flash? MCUboot doesn't really know if the flash is internal or external, that's up to the OS backend, but I am wondering if there is some pattern here. How many upgrades in average I would have to run to see the issue? Do you know if uploading new firmware with MCUmgr in between tests is necessary to trigger the issue, or just doing test/revert of the same images is enough?

Comment (nordicjm) [reply docs/readme-zephyr.md:211]: Not sure on that, it was completely random as to when it happened, as bad as it sounds there was no easily reproducible method to get a device into this state (it was all during development, might have been reported post production too but I do not remember) - images would have been loaded via MCUmgr (bluetooth and/or shell transports) then marked as test from that, the image wasn't left and flipped between test and none.

Comment (d3zd3z) [reply docs/readme-zephyr.md:211]: Do we need to create an issue to track this? Ideally, we would be able to figure out how to reproduce this in the simulator. Does it happen only with the write sizes differing between internal and external? But, I don't think creating the issue should block merging this issue. Let me know in a comment that there is an issue created, and this can proceed.

Comment (nordicjm) [reply docs/readme-zephyr.md:211]: I gave it a quick try to reproduce it and was not able to in 20 minutes or so. The write and erase sizes for both memories were the same (mx25... QSPI flash - 4KB sector/erase size, 4 byte minimum write size, nrf52840 flash - 4KB sector/erase size, 4 byte minimum write size), and it was completely random as to how it got into that mode, but once it was in that mode, you were unable to get it out of that mode or load a new update, mcuboot would not do any swapping and the app would not allow uploading a new image because the image was marked for test

Comment (utzig) [reply docs/readme-zephyr.md:211]: Yes, please open an issue with the same description. I have used swap with scratch and newtmgr together for many years with Mynewt, and to some extent with Zephyr as well, and never bumped into this bug. I am somewhat suspicious it might be related to the QSPI driver, like an error write that does not raise an error or something similar, but it's just a guess. Btw, I've never tried scratch on the QSPI flash, but for MCUboot flash is just flash, wherever it is located...

**Comment by d3zd3z** [comment]
I'll approve this once the tracking issue is created for the problem described about unreliability. Please reference the issue in a comment here.

**Review by d3zd3z — APPROVED** [review]

**Pull request merged**
